### PR TITLE
Handle cairo1 failed tests

### PR DESCRIPTION
**Pull request by karol-bisztyga**

### CODE REVIEW

**Review by piotmag769 — CHANGES_REQUESTED** [review]

**Review by Arcticae — APPROVED** [review]
LGTM
(but please adjust the tests before merging)

File: tests/integration/cairo1_compiler_bindings/library_functions_test.py [hunk 1]
```diff
@@ -67,9 +67,7 @@ def check_library_function(
                 },
             )
 
-            assert cairo_runner_facade.get_return_values(3)[0] == int(
-                bool(mocked_error_code)
-            )
+            assert cairo_runner_facade.did_panic() == int(bool(mocked_error_code))
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/cairo1_compiler_bindings/library_functions_test.py]: [remove redundant cast](https://github.com/software-mansion/protostar/pull/1572/commits/51c097dff592cfc2d7954b8424cec4cb03bab939)

File: tests/integration/pure_cairo_vm/cairo1_integration/cairo1_runner_test.py [hunk 2]
```diff
@@ -162,3 +162,33 @@ async def test_cairo_1_runner_single_case(
...
+    )
+    test_path_str = str(datadir / "test_cairo1_failing.cairo")
+
+    targets = {
+        "test_ok": True,
+        "test_panic_single_value": False,
+        "test_panic_multiple_values": False,
+    }
+    for test_target, success in targets.items():
+        testing_summary = await protostar.run_test_runner(
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/pure_cairo_vm/cairo1_integration/cairo1_runner_test.py]: [run as one suite](https://github.com/software-mansion/protostar/pull/1572/commits/9af8a6f1f53b1e0692595602c5c79f769743e7c8)

File: tests/integration/cairo1_compiler_bindings/return_values_test.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        },
+        "test::test::test_panic_single_value": {
+            "did_panic": True,
+            "panic_data": [21],
+        },
+        "test::test::test_panic_multiple_values": {
+            "did_panic": True,
+            "panic_data": [101, 102, 103],
+        },
+    }
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/cairo1_compiler_bindings/return_values_test.py]: As discussed offline, we decided not to do this.

File: tests/integration/cairo1_compiler_bindings/return_values_test.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    actual_outcome = {}
+    for test_case_name, offset in protostar_casm.offset_map.items():
+        cairo_runner_facade.run_from_offset(offset=offset)
+        cairo_runner_facade.did_panic()
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply tests/integration/cairo1_compiler_bindings/return_values_test.py]: [remove redundant line](https://github.com/software-mansion/protostar/pull/1572/commits/e70dc63a7798933cbc7b35f928a534070535ec1d)

File: tests/e2e/cairo1/test_cairo1_testing.py [hunk 5]
```diff
@@ -16,3 +18,28 @@ def test_no_tests_found(protostar: ProtostarFixture):
...
+        "[PASS] tests/failing_test.cairo test_ok",
+        "[FAIL] tests/failing_test.cairo test_panic_multiple_values",
+        "Test failed with data: [101, 102, 103]",
+        "[FAIL] tests/failing_test.cairo test_panic_single_value",
+        "Test failed with data: [21]",
+        "1 failed, 1 total",
+        "2 failed, 1 passed, 3 total",
+    ]
+
+    for expected_output_line in expected_output_lines:
```
Comment (Arcticae) [inline tests/e2e/cairo1/test_cairo1_testing.py:44]: [nitpick] I actually don't think we need to assert the whole content, just a few bits would be okay (is the failure data included, are the functions collected, how many have failed and passed)

This is fine as-is though

File: tests/e2e/cairo1/test_cairo1_testing.py [hunk 6]
```diff
@@ -16,3 +18,28 @@ def test_no_tests_found(protostar: ProtostarFixture):
     result = protostar(["test-cairo1", "tests"])
 
     assert "No test cases found" in result
+
+
+def test_failing_tests(protostar: ProtostarFixture, copy_fixture: CopyFixture):
+    copy_fixture("cairo1_project", "./cairo1_project")
+    template_test_files = Path("cairo1_project/tests").glob("**/*")
+    for template_test_file in template_test_files:
```
Comment (Arcticae) [inline tests/e2e/cairo1/test_cairo1_testing.py:26]: Perhaps this should use a version of protostar fixture which initializes a cairo1 project? Is it possible?

**Review by Arcticae — APPROVED** [review]
LGTM, e2e test could be improved but not a blocking thing tbh

File: tests/e2e/cairo1/test_cairo1_testing.py (hunk 5 above)
Comment (karol-bisztyga) [reply tests/e2e/cairo1/test_cairo1_testing.py:44]: Hmm, I think I covered pretty much what you mentioned + what failed and what passed, I think it's okay, it's not a big overhead or something.

File: tests/e2e/cairo1/test_cairo1_testing.py (hunk 6 above)
Comment (piotmag769) [reply tests/e2e/cairo1/test_cairo1_testing.py:26]: @MaksymilianDemitraszek was against this approach afaik

Comment (karol-bisztyga) [reply tests/e2e/cairo1/test_cairo1_testing.py:26]: It is a cairo1 project, I just removed the template tests as they don't check what we need here. If you feel like we should include them anyway, please let me know but I think we're good without 'em.

**Review by piotmag769 — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

**Pull request merged**